module/os/linux/spl/spl-taskq.c

```diff
@@ -1255,12 +1284,101 @@ module_param_call(spl_taskq_kick, param_set_taskq_kick, param_get_uint,
 MODULE_PARM_DESC(spl_taskq_kick,
 	"Write nonzero to kick stuck taskqs to spawn more threads");
 
+/*
+ * This callback will be called exactly once for each core that comes online,
+ * for each dynamic taskq.  There are two kinds of dynamic taskqs that we
+ * attempt to expand; those who have TASKQ_THREADS_CPU_PCT set, and those
+ * created with the same number of threads as we have cores. In the CPU_PCT
+ * case, we need to redo the percentage calculation each time this is
+ * called. For the threads == cores case, we always want to create a new
+ * thread to keep the number of cores and threads in sync.
+ */
+static int
+spl_taskq_expand(unsigned int cpu, struct hlist_node *node)
```

Comment:
You'll need to wrap these functions as well to avoid some compiler warnings on older kernels.

```
.../os/linux/spl/spl-taskq.c:1297:1: error: ‘spl_taskq_expand’ defined but not used [-Werror=unused-function]
 spl_taskq_expand(unsigned int cpu, struct hlist_node *node)
 ^
.../os/linux/spl/spl-taskq.c:1334:1: error: ‘spl_taskq_prepare_down’ defined but not used [-Werror=unused-function]
 spl_taskq_prepare_down(unsigned int cpu, struct hlist_node *node)
```